### PR TITLE
Refactoring tasks

### PR DESCRIPTION
**Pull request by danaack**
Changes include:
-move kv_io.h code into kv_systems
-remove all reference and instances of kv_io.h and kv_io.c
-update dictionary_size to be of type unsigned int

### CODE REVIEW

**Comment by Stickerpants** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;jbwizufW&#x2F;230-fix-and-refactor-kv-io-h"> Fix and refactor kv_io.h</a>

File: src/cpp_wrapper/Dictionary.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ ion_dictionary_handler_t	handler;
 ion_dictionary_t			dict;
 int							size_k;
 int							size_v;
-int							dict_size;
+unsigned int				dict_size;
```
Comment (Stickerpants) [inline src/cpp_wrapper/Dictionary.h]: Ideally, these would all use `ion_dictionary_size_t` - how feasible would that be to do?

**Comment by Stickerpants** [comment]
LGTM

**Comment by Stickerpants** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Y9tOef7I&#x2F;232-change-types-that-should-be-unsigned-as-appropriate"> Change types that should be unsigned as appropriate</a>

**Comment by graemedouglas** [comment]
LGTM

**Pull request merged**